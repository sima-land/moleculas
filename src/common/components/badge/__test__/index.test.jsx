import React from 'react';
import { mount } from 'enzyme';
import { Badge } from '..';
import { addDays, addMonths } from 'date-fns';

describe('<Badge />', () => {
  beforeEach(() => {
    jest.useFakeTimers('modern');
    jest.setSystemTime(new Date('2021-01-01T09:00:00'));
  });

  it('should renders correctly', () => {
    const variants = [
      // text view
      {
        color: '#00b8d4',
        href: 'https://sima-land.ru',
        fields: [{ type: 'text', value: 'Система скидок' }],
      },

      // icon only view
      {
        color: '#ff7200',
        href: 'https://sima-land.ru',
        fields: [
          {
            type: 'svg-url',
            value: 'https://static2.static1-sima-land.com/image/mobile_app/common/notice_icon.svg',
          },
        ],
      },

      // timer view
      {
        color: '#b52ea8',
        href: 'https://sima-land.ru',
        fields: [{ type: 'timer', value: addDays(new Date(), 7).toISOString() }],
      },

      // timer + text view
      {
        color: '#b52ea8',
        href: 'https://sima-land.ru',
        fields: [
          { type: 'timer', value: addMonths(new Date(), 3).toISOString() },
          { type: 'text', value: ' — Специальные цены на музыкальные инструменты' },
        ],
      },

      // no hover effect
      {
        color: '#607d8b',
        fields: [{ type: 'text', value: 'Уценённый товар' }],
      },
    ];

    variants.forEach(variant => {
      const wrapper = mount(<Badge {...variant} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
