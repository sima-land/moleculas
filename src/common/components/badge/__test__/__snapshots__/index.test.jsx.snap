// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should renders correctly 1`] = `
<Badge
  color="#00b8d4"
  fields={
    Array [
      Object {
        "type": "text",
        "value": "Система скидок",
      },
    ]
  }
  href="https://sima-land.ru"
>
  <a
    className="root interactive"
    href="https://sima-land.ru"
    style={
      Object {
        "--badge-color": "#00b8d4",
      }
    }
  >
    <span
      className="content"
    >
      Система скидок
    </span>
  </a>
</Badge>
`;

exports[`<Badge /> should renders correctly 2`] = `
<Badge
  color="#ff7200"
  fields={
    Array [
      Object {
        "type": "svg-url",
        "value": "https://static2.static1-sima-land.com/image/mobile_app/common/notice_icon.svg",
      },
    ]
  }
  href="https://sima-land.ru"
>
  <a
    className="root icon-only interactive"
    href="https://sima-land.ru"
    style={
      Object {
        "--badge-color": "#ff7200",
      }
    }
  >
    <img
      className="icon"
      src="https://static2.static1-sima-land.com/image/mobile_app/common/notice_icon.svg"
    />
  </a>
</Badge>
`;

exports[`<Badge /> should renders correctly 3`] = `
<Badge
  color="#b52ea8"
  fields={
    Array [
      Object {
        "type": "timer",
        "value": "2021-01-08T09:00:00.000Z",
      },
    ]
  }
  href="https://sima-land.ru"
>
  <a
    className="root interactive"
    href="https://sima-land.ru"
    style={
      Object {
        "--badge-color": "#b52ea8",
      }
    }
  >
    <span
      className="content"
    >
      <Timer
        date="2021-01-08T09:00:00.000Z"
        format={[Function]}
        key="0"
        timeout={60000}
      >
        07:00:00
      </Timer>
    </span>
  </a>
</Badge>
`;

exports[`<Badge /> should renders correctly 4`] = `
<Badge
  color="#b52ea8"
  fields={
    Array [
      Object {
        "type": "timer",
        "value": "2021-04-01T09:00:00.000Z",
      },
      Object {
        "type": "text",
        "value": " — Специальные цены на музыкальные инструменты",
      },
    ]
  }
  href="https://sima-land.ru"
>
  <a
    className="root interactive"
    href="https://sima-land.ru"
    style={
      Object {
        "--badge-color": "#b52ea8",
      }
    }
  >
    <span
      className="content"
    >
      <Timer
        date="2021-04-01T09:00:00.000Z"
        format={[Function]}
        key="0"
        timeout={60000}
      >
        90:00:00
      </Timer>
       
       — Специальные цены на музыкальные инструменты
    </span>
  </a>
</Badge>
`;

exports[`<Badge /> should renders correctly 5`] = `
<Badge
  color="#607d8b"
  fields={
    Array [
      Object {
        "type": "text",
        "value": "Уценённый товар",
      },
    ]
  }
>
  <span
    className="root"
    style={
      Object {
        "--badge-color": "#607d8b",
      }
    }
  >
    <span
      className="content"
    >
      Уценённый товар
    </span>
  </span>
</Badge>
`;
